src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Trait that allows a generic method to find and sort service by priority option in the tag.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+trait PriorityTaggedServiceTrait
+{
+    /**
+     * Finds all services with the given tag name and order them by their priority.
+     *
+     * @param string           $tagName
+     * @param ContainerBuilder $container
+     *
+     * @return Reference[]
+     */
+    private function findAndSortTaggedServices($tagName, ContainerBuilder $container)
+    {
+        $services = $container->findTaggedServiceIds($tagName);
+
+        $queue = new \SplPriorityQueue();
+
```

Comment:
Just to be sure (ref https://github.com/symfony/symfony/pull/18450#issuecomment-213767566).

This PR make some passes use a trait to collect tagged services, are concerned:

1. `Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface`

2. `Symfony\Component\Config\Resource\ResourceInterface\ResourceCheckerInterface`

3. `Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface`

4. `Symfony\Component\PropertyInfo\PropertyAccessExtractorInterface`

5. `Symfony\Component\Serializer\Encoder\EncoderInterface`

Excepted 1 and 4, these interfaces have kind of a `supports` method (serializer encoders have `supportsEncoding`).

After a deep look at each of them, I ask again, does it really make sense to allow duplicated class with different IDs or with different priorities? Should't this be tested?

Or is it the responsibility of each service using them to perform that check? Because currently, it seems nothing is preventing it.